src/setup.erl

```diff
@@ -132,18 +139,27 @@ enable_cluster_int(Options, no) ->
           Pw -> Pw
         end
     },
+    ok = require_admins(CurrentAdmins, NewCredentials),
 
     % if bind_address == 127.0.0.1 and no bind_address in req -> error
     CurrentBindAddress = config:get("chttpd","bind_address"),
     NewBindAddress = proplists:get_value(bind_address, Options),
-    ok = require_admins(CurrentAdmins, NewCredentials),
     ok = require_bind_address(CurrentBindAddress, NewBindAddress),
 
+    NodeCount = couch_util:get_value(node_count, Options),
+    ok = require_node_count(NodeCount),
+
+    Port = proplists:get_value(port, Options),
+
+    setup_node(NewCredentials, NewBindAddress, NodeCount, Port),
+    couch_log:notice("Enable Cluster: ~p~n", [Options]).
+
+setup_node(NewCredentials, NewBindAddress, NodeCount, Port) ->
```

Comment:
Refactored the work of _actually_ doing the setup so it can be used by `enable_cluster_int` as well as `enable_single_node`.